src/server/route/routes/user.ts

```diff
@@ -0,0 +1,110 @@
+import { router } from '../router';
+import { Request, Response } from 'express';
+import { db } from '../../dbConfigs';
+import { Login, Registration } from '../../../../types';
+import bcrypt from 'bcrypt';
+import * as jwt from 'jsonwebtoken';
+import dotenv from 'dotenv';
+dotenv.config();
+
+router.route('/user/register').post(async (req: Request, res: Response) => {
+    const user: Registration = req.body;
+
+    if (!user || !user.username || !user.password || !user.email) {
+        res.status(403).json({ message: 'Missing parameters' });
+        return;
+    }
+
+    const saltRounds = 10;
+    const hash = await bcrypt.hash(user.password, saltRounds);
+    const q = await db.query(
+        'SELECT username FROM users WHERE username=$1 OR email=$2',
+        [user.username, user.email]
+    );
+    if (q.rowCount == 0) {
+        await db.query(
+            'INSERT INTO users (username, password, email) VALUES ($1, $2, $3)',
+            [user.username, hash, user.email]
+        );
+        res.status(200).json();
+    } else {
+        res.status(403).json({
+            message: 'Username or email already registered',
+        });
+    }
+});
+
+router.route('/user/login').post(async (req: Request, res: Response) => {
+    const body: Login = req.body;
+    let user = null;
+
+    if (!body.password || (!body.email && !body.username)) {
+        res.status(403).json({ message: 'Missing parameters' }).end();
+        return;
+    }
+
+    //User tries to login with email
+    if (body.email && body.email.includes('@')) {
+        const { rows, rowCount } = await db.query(
+            'SELECT * FROM users WHERE email=$1',
+            [body.email]
+        );
+
+        if (rowCount == 0) {
+            res.status(401).json({ message: 'Wrong email or password' }).end();
+            return;
+        } else if (rowCount >= 2) {
```

Comment:
This is defensive programming, which is an antipattern. It's better to ensure that email is an unique field, so that selecting from it will always return 0 or 1 rows.